common/djangoapps/enrollment/views.py

```diff
@@ -469,6 +472,14 @@ def get(self, request):
                     ).format(username=username)
                 }
             )
+        if username == request.user.username or self.has_api_key_permissions(request):
+            return Response(enrollment_data)
+        filtered_data = []
+        for enrollment in enrollment_data:
```

Comment:
Hi All!

Yes, getting all courses is quite slow currently and the courses-via-overview PR referenced will certainly speed it up.  I'm not sure when that PR will be merged though, but should have an estimate later today or the next.

I'm not filled in on the details on getting `enrollment_data`, but if that is quick, you'll be able to loop through the courses returned (assuming that is a complete list) and use `courseware.access.has_access` to determine access.  This is probably the better way to do this because the access logic will provide organizational access (e.g. `OrgStaffRole`) and instructor (a.k.a. admin) access.  I suggest adding tests for instructor/admin and and organizational level access too if you're supporting those cases.

As for moving `find_courses` up a level, there is a tiny bit of complexity around cached results, but it's certainly feasible.  I do think if you have the courses already, then using `has_access` should be fairly quick.